pkg/controller/nodeipam/ipam/cidrset/metrics.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cidrset
+
+import (
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const rangeAllocatorSubsystem = "range_allocator"
+
+var (
+	cidrSetAllocations = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Subsystem:      rangeAllocatorSubsystem,
+			Name:           "cidrset_allocation_total",
+			Help:           "Counter measuring total number of CIDR allocations.",
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{"clusterCIDR"},
+	)
+	cidrSetReleases = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Subsystem:      rangeAllocatorSubsystem,
+			Name:           "cidrset_release_total",
+			Help:           "Counter measuring total number of CIDR releases.",
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{"clusterCIDR"},
+	)
+	cidrSetRetries = metrics.NewCounterVec(
```

Comment:
I', not so sure this one is useful, given the histogram - convince me?

Reply:
you are right, it creates automatically y the counters https://prometheus.io/docs/practices/histograms/ and this is not needed.
These things are the ones that make you feel old, where are those days parsing rrds with perl and exposing it with apache perl-cgi :)